lib/prune.js

```diff
@@ -21,6 +21,8 @@ prune.completion = require('./utils/completion/installed-deep.js')
 
 function prune (args, cb) {
   var dryrun = !!npm.config.get('dry-run')
+  // prune should not touch the lock file
+  npm.config.set('save', false)
```

Comment:
I think `npm.config.set('shrinkwrap', false)` does not work because we need to avoid `saveToDependencies` at [lib/install.js#L294-L300](https://github.com/npm/npm/blob/90a25f65804d7d7dc351d01c53826b879139d8da/lib/install.js#L294-L300) to not make the lock file with `npm prune`, and the flag comes from here [lib/install/save.js#L133-L136](https://github.com/npm/npm/blob/latest/lib/install/save.js#L133-L136). The default value of the save option is `true` since npm@5, then we can turn off at `npm prune`.

`npm.config.set` is to set a value to a config property in runtime. It doesn't overwrite `.npmrc`.